ethstats/ethstats.go

```diff
@@ -123,17 +124,34 @@ func (s *Service) loop() {
 
 	// Loop reporting until termination
 	for {
-		// Establish a websocket connection to the server and authenticate the node
+		// Resolve the URL, defaulting to TLS, but falling back to none too
 		url := fmt.Sprintf("%s/api", s.host)
+
+		urls := []string{url}
 		if !strings.Contains(url, "://") {
-			url = "wss://" + url
+			urls = []string{"wss://" + url, "ws://" + url}
```

Comment:
Boom. This blew up. IsAbs returns true if I have a port number specified https://play.golang.org/p/l8SbjsQslG